kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/Euclidean2DSpace.kt

```diff
@@ -1,20 +1,21 @@
 package scientifik.kmath.geometry
 
+import scientifik.kmath.dimensions.D2
 import scientifik.kmath.linear.Point
+import scientifik.kmath.operations.ExtendedField
+import scientifik.kmath.operations.RealField
 import scientifik.kmath.operations.SpaceElement
-import scientifik.kmath.operations.invoke
 import kotlin.math.sqrt
 
-
-interface Vector2D : Point<Double>, Vector, SpaceElement<Vector2D, Vector2D, Euclidean2DSpace> {
-    val x: Double
-    val y: Double
-
+data class Vector2D(
```

Comment:
I am not sure I like this change. We may lose compatibility with external geometry libraries. This should be discussed.